Update documentation of the matching Adyen Notification with commercetools payment. (#503)
commercetools/commercetools-adyen-integration: extension/docs/WebComponentsIntegrationGuide.md

}
```

> If you are [creating a commercetools payment](https://docs.commercetools.com/http-api-projects-payments#create-a-payment), the payment draft have to contain the `paymentMethodInfo.paymentInterface = ctp-adyen-integration` and `amountPlanned` value, for example:

Author: https://github.com/commercetools/commercetools-adyen-integration/pull/503#discussion_r551347421

After the shopper submits their payment details or chooses to pay with a payment method that requires a redirection,
the Adyen Web Components will generate a `makePaymentRequest`. 

**Preconditions:**

> - `makePaymentRequest` must contain a unique payment `reference` value. The reference value cannot be duplicated in any commercetools payment and it's a required field by Adyen. The extension module uses reference value to set payment key, later it acts as a unique link between commercetools payment and Adyen payment(`merchantReference`). Reference may only contain alphanumeric characters, underscores, and hyphens and must have a minimum length of 2 characters and a maximum length of 80 characters. 

Author: Good comment, I am also considering the requirements of the Adyen: https://docs.adyen.com/api-explorer/#/CheckoutService/v66/post/payments__reqParam_reference

What is your suggestion here?

> - `makePaymentRequest` must contain a unique payment `reference` value. The reference value cannot be duplicated in any commercetools payment and it's a required field by Adyen. The extension module uses reference value to set payment key, later it acts as a unique link between commercetools payment and Adyen payment(`merchantReference`). Reference may only contain alphanumeric characters, underscores, and hyphens and must have a minimum length of 2 characters and a maximum length of 80 characters. 

Reviewer: For now we can just leave it in the docs. If users don't follow the rules, it's their fault :D

After the shopper submits their payment details or chooses to pay with a payment method that requires a redirection,
the Adyen Web Components will generate a `makePaymentRequest`. 

**Preconditions:**

> - `makePaymentRequest` must contain a unique payment `reference` value, the reference value cannot be re-used for other commercetools payment and it's a required field by Adyen. The extension module uses reference value to set payment key, later it acts as a unique link between commercetools payment and Adyen payment. Reference may only contain alphanumeric characters, underscores, and hyphens and must have a minimum length of 2 characters and a maximum length of 80 characters. 

Reviewer 2: ```suggestion
- `makePaymentRequest` must contain a unique payment `reference` value. The reference value cannot be duplicated in any commercetools payment and it's a required field by Adyen. 

The extension module uses reference value to set payment key, later it acts as a unique link between commercetools payment and Adyen payment(`merchantReference`). Reference may only contain alphanumeric characters, underscores, and hyphens and must have a minimum length of 2 characters and a maximum length of 80 characters. 
```

    {
      "action": "setCustomField",
      "name": "makePaymentRequest",
      "value": "{\"amount\":{\"currency\":\"EUR\",\"value\":1000},\"reference\":\"YOUR_REFERENCE\",\"paymentMethod\":{\"type\":\"scheme\",\"encryptedCardNumber\":\"test_4111111111111111\",\"encryptedExpiryMonth\":\"test_03\",\"encryptedExpiryYear\":\"test_2030\",\"encryptedSecurityCode\":\"test_737\"},\"returnUrl\":\"https://your-company.com/...\",\"merchantAccount\":\"YOUR_MERCHANT_ACCOUNT\"}"
    }
  ]
}
```

> If you are [creating a new commercetools payment](https://docs.commercetools.com/http-api-projects-payments#create-a-payment), the payment draft have to contain the `paymentMethodInfo.paymentInterface = ctp-adyen-integration` and `amountPlanned` value additional to the `makePaymentRequest` custom field, for example:

Author: Yes, but I think it's ok to explain it again because this the first place in the integration that we are explaining a new payment draft. Is it ok to add only the link? Or what would be best?

> If you are [creating a new commercetools payment](https://docs.commercetools.com/http-api-projects-payments#create-a-payment), the payment draft have to contain the `paymentMethodInfo.paymentInterface = ctp-adyen-integration` and `amountPlanned` value additional to the `makePaymentRequest` custom field, for example:

Reviewer: I agree, just a link is ok :)

> If you are [creating a new commercetools payment](https://docs.commercetools.com/http-api-projects-payments#create-a-payment), the payment draft have to contain the `paymentMethodInfo.paymentInterface = ctp-adyen-integration` and `amountPlanned` value additional to the `makePaymentRequest` custom field, for example:

Author: I think this is a different part, creating a new object is not related to the validation section, I could not rephrase the sentence with this validation link.

> If you are [creating a new commercetools payment](https://docs.commercetools.com/http-api-projects-payments#create-a-payment), the payment draft have to contain the `paymentMethodInfo.paymentInterface = ctp-adyen-integration` and `amountPlanned` value additional to the `makePaymentRequest` custom field, for example:

Reviewer: ```suggestion
If you are [creating a new commercetools payment](https://docs.commercetools.com/http-api-projects-payments#create-a-payment), the payment draft have to contain the `paymentMethodInfo.paymentInterface = ctp-adyen-integration` (see https://github.com/commercetools/commercetools-adyen-integration/blob/master/extension/docs/WebComponentsIntegrationGuide.md#validate-payment) and `amountPlanned` value additional to the `makePaymentRequest` custom field, for example:
```

````json
{
  "amountPlanned": {
    "currencyCode": "EUR",
    "centAmount": 1000
  },
  "paymentMethodInfo": {
    "paymentInterface": "ctp-adyen-integration"
  },
  "custom": {
    "type": {
      "typeId": "type",
      "key": "ctp-adyen-integration-web-components-payment-type"
    },
    "fields": {
      "makePaymentRequest": "{\"amount\":{\"currency\":\"EUR\",\"value\":1000},\"reference\":\"YOUR_REFERENCE\",\"paymentMethod\":{\"type\":\"scheme\",\"encryptedCardNumber\":\"test_4111111111111111\",\"encryptedExpiryMonth\":\"test_03\",\"encryptedExpiryYear\":\"test_2030\",\"encryptedSecurityCode\":\"test_737\"},\"returnUrl\":\"https://your-company.com/...\",\"merchantAccount\":\"YOUR_MERCHANT_ACCOUNT\"}"
    }
  }
}
````

> The response from Adyen is added to `makePaymentResponse` custom field. 

Author: https://github.com/commercetools/commercetools-adyen-integration/pull/503/commits/eb9ffc3192a6235e53bc0c8f1b4445616c61e077
